rfd/xxxx-tsh-bench-db.md

```diff
@@ -0,0 +1,166 @@
+---
+authors: Gabriel Corado (gabriel.oliveira@goteleport.com)
+state: draft
+---
+
+# RFD xxxx - tsh bench db
+
+## Required Approvals
+
+* Engineering: @smallinsky && @r0mant
+* Product: @xinding33 || @klizhentas
+* Security: @reedloden
+
+## What
+Introduce a new subcommand, `tsh bench db`, which will work similarly to
+`tsh bench` for SSH and Kubernetes benchmarking but for database access.
+
+## Why
+The team is working towards understanding performance issues with database
+access. This built-in tool will be used to generate database access load and
+produce some client-side metrics that will be used to understand overall
+performance better.
```

Comment:
How to test "local proxy" connections and "local proxy tunnelled" connections (as opposed to "direct Proxy" connections)? Postgres for example, the flows for these connection modes will differ on both client and server side. If not planned for the first version, maybe consider in future plan.

Reply:
I have yet to elaborate on this since I've kept it outside the initial version. Most of my initial work was to mimic the `tsh connect` flow for PostgreSQL and MySQL inside the tests.

Do you think we should already cover this? Also, @smallinsky, do you have any thoughts on this?

Reply:
> Do you think we should already cover this? Also, @smallinsky, do you have any thoughts on this?

Agree, but the tsh bench command should be Teleport Proxy independed and probably in first version it should work with in Our Cloud Env where TLS Routing is enabled and separate MySQL/Postgres port configuration is deprecated.  So i think that it should be nice to cover multiplex mode too at first implementation or cover only multiplex mode.